r-package/grf/vignettes/llf.Rmd

```diff
@@ -0,0 +1,147 @@
+preds.llf <- results.llf$predictions
+mean((preds.llf - truth)**2)
+
+# llf predictions from ll_regression_forest
+forest <- ll_regression_forest(X, Y, honesty = TRUE)
+results.llf <- predict(forest, X.test) 
+preds.llf <- results.llf$predictions
+mean((preds.llf - truth)**2)
+```
+
+When we perform LLF predictions, we can either do a standard ridge regression (ll.weight.penalty set to FALSE), or scale by the covariance matrix (ll.weight.penalty set to TRUE). This defaults to FALSE.
+
+```{r}
+results.llf.weighted <- predict(forest, X.test, 
+                               ll.weight.penalty = TRUE) 
+preds.llf.weighted <- results.llf.weighted$predictions
+mean((preds.llf.weighted - truth)**2)
+```
+
+The user can choose to specify a ridge regression parameter ll.lambda. This defaults to NULL, and if not modified will be selected by automatic parameter tuning. In general, we recommend letting the forest tune this parameter, or performing your own cross-validation loop. The exception to this would be for very large datasets. 
+
+```{r}
+results.llf.lambda <- predict(forest, X.test, 
+                             ll.lambda = 0.1) 
+preds.llf.lambda <- results.llf.lambda$predictions
+mean((preds.llf.lambda - truth)**2)
+```
+
+We also consider the role of tree training in local linear forests. A standard CART split minimizes prediction error from predicting leaf-wide averages. Instead, we can use residual splits, which minimize the corresponding prediction errors on ridge regression residuals. This is currently an experimental feature.  
+
+```{r}
+forest <- ll_regression_forest(X, Y)
+preds.cart.splits <- predict(forest, X.test)
+
+ll.forest <- ll_regression_forest(X, Y, enable.ll.split = TRUE)
+preds.ll.splits <- predict(ll.forest, X.test)
+
+mse.cart.splits <- mean((preds.cart.splits$predictions - truth)^2)
+mse.ll.splits <- mean((preds.ll.splits$predictions - truth)^2)
+
+mse.cart.splits
+mse.ll.splits
+```
+
+Especially with many covariates, it is reasonable to restrict the local regression to only include a few features of interest. We recommend using the lasso. 
+
+```{r}
+# Select covariates 
+lasso.mod <- cv.glmnet(X, Y, alpha = 1)
+lasso.coef <- predict(lasso.mod, type = "nonzero")
+selected <- ifelse(is.null(dim(lasso.coef)), 1:ncol(X), lasso.coef[,1])
+
+# Train LLF
+llf.lasso.preds <- predict(forest, X.test, 
+                           linear.correction.variables = selected,
+                           ll.weight.penalty = TRUE)
+results.llf.lasso <- llf.lasso.preds$predictions
+mean((results.llf.lasso - truth)**2)
+```
+
+Last, consider variance estimates and confidence intervals. A user must specify estimate.variance to be TRUE. A parameter under the hood is ci.group.size, which must be at least 2 in order to grow variance estimates (defaults to 2); the forest will grow ci.group.size trees on every subsample in order to provide confidence intervals. We use a different data-generating process for easier visualization.
```

Comment:
I think this is not different from the usual GRF setting, so you can probably cut this in the interest of making this about LLFs only.